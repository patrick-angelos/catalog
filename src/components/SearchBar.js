import React from 'react';
import PropTypes from 'prop-types';
import {
  cuisineValues, dishValues, mealValues,
} from '../assets/logic/filterValues';
import Option from './Option';

const SearchBar = (props) => {
  const {
    filters, link, filterHandler, submitHandler, clear,
  } = props;

  const handleFilterChange = (e) => {
    const filter = e.target.id;
    const defaults = ['Cuisine', 'Dish', 'Meal'];
    const value = defaults.includes(e.target.value) ? '' : e.target.value;
    filterHandler(filter, value);
  };

  const handleSubmit = () => {
    submitHandler();
  };

  const handleClear = () => {
    clear();
  };

  const mapOptions = (values) => {
    const options = values.map((option) => <option key={option}>{option}</option>);
    return options;
  };

  const cuisineOptions = mapOptions(cuisineValues);
  const dishOptions = mapOptions(dishValues);
  const mealOptions = mapOptions(mealValues);

  return (
    <div id="search" className="search-area">
      <div
        className="filters background-black"
        onChange={(e) => handleFilterChange(e)}
      >
        <input className="search-bar" id="q" type="text" placeholder="Search" defaultValue={filters.q} />
        <div className="selection-wrapper flex space-between">
          <Option id="cuisineType" options={cuisineOptions} value={filters.cuisineType} />
          <Option id="dishType" options={dishOptions} value={filters.dishType} />
          <Option id="mealType" options={mealOptions} value={filters.mealType} />
        </div>
        <a href={link} className="submit background-white sans" type="button" onClick={handleSubmit}>Search</a>
        <button className="submit background-white sans" type="button" onClick={handleClear}>Reset filters</button>
      </div>
    </div>
  );
};

SearchBar.propTypes = {
  filters: PropTypes.shape({
    q: PropTypes.string.isRequired,
    cuisineType: PropTypes.string.isRequired,
    dishType: PropTypes.string.isRequired,
    mealType: PropTypes.string.isRequired,
  }).isRequired,
  link: PropTypes.string,
  filterHandler: PropTypes.func.isRequired,
  submitHandler: PropTypes.func,
  clear: PropTypes.func.isRequired,
};

SearchBar.defaultProps = {
  link: '#search',
  submitHandler: () => {},
};

export default SearchBar;
