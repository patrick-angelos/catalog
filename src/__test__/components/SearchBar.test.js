import React from 'react';
import renderer from 'react-test-renderer';
import SearchBar from '../../components/SearchBar';

describe('SearchBar', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<SearchBar
        filters={{
          q: '', cuisineType: '', dishType: '', mealType: '',
        }}
        filterHandler={() => {}}
        clear={() => {}}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
