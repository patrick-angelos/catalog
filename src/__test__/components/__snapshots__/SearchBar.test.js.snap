// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly 1`] = `
<div
  className="search-area"
  id="search"
>
  <div
    className="filters background-black"
    onChange={[Function]}
  >
    <input
      className="search-bar"
      defaultValue=""
      id="q"
      placeholder="Search"
      type="text"
    />
    <div
      className="selection-wrapper flex space-between"
    >
      <div
        className="option-wrapper flex space-between relative"
      >
        <select
          className="option"
          defaultValue=""
          id="cuisineType"
        >
          <option>
            Cuisine
          </option>
          <option>
            American
          </option>
          <option>
            Asian
          </option>
          <option>
            British
          </option>
          <option>
            Caribbean
          </option>
          <option>
            Central Europe
          </option>
          <option>
            Chinese
          </option>
          <option>
            Eastern Europe
          </option>
          <option>
            French
          </option>
          <option>
            Indian
          </option>
          <option>
            Italian
          </option>
          <option>
            Japanese
          </option>
          <option>
            Kosher
          </option>
          <option>
            Mediterranean
          </option>
          <option>
            Mexican
          </option>
          <option>
            Middle Eastern
          </option>
          <option>
            Nordic
          </option>
          <option>
            South American
          </option>
          <option>
            South East Asian
          </option>
        </select>
        <div
          className="arrow absolute"
        />
      </div>
      <div
        className="option-wrapper flex space-between relative"
      >
        <select
          className="option"
          defaultValue=""
          id="dishType"
        >
          <option>
            Dish
          </option>
          <option>
            alcohol
          </option>
          <option>
            biscuits and cookies
          </option>
          <option>
            bread
          </option>
          <option>
            cereals
          </option>
          <option>
            condiments and sauces
          </option>
          <option>
            drinks
          </option>
          <option>
            desserts
          </option>
          <option>
            egg
          </option>
          <option>
            main course
          </option>
          <option>
            omelet
          </option>
          <option>
            pancake
          </option>
          <option>
            preps
          </option>
          <option>
            preserve
          </option>
          <option>
            salad
          </option>
          <option>
            sandwiches
          </option>
          <option>
            soup
          </option>
          <option>
            starter
          </option>
        </select>
        <div
          className="arrow absolute"
        />
      </div>
      <div
        className="option-wrapper flex space-between relative"
      >
        <select
          className="option"
          defaultValue=""
          id="mealType"
        >
          <option>
            Meal
          </option>
          <option>
            Breakfast
          </option>
          <option>
            Lunch
          </option>
          <option>
            Dinner
          </option>
          <option>
            Snack
          </option>
          <option>
            Teatime
          </option>
        </select>
        <div
          className="arrow absolute"
        />
      </div>
    </div>
    <a
      className="submit background-white sans"
      href="#search"
      onClick={[Function]}
      type="button"
    >
      Search
    </a>
    <button
      className="submit background-white sans"
      onClick={[Function]}
      type="button"
    >
      Reset filters
    </button>
  </div>
</div>
`;
